assets/src/scripts/store/instantaneous-value-time-series-state.js

```diff
@@ -0,0 +1,295 @@
+ * in cursor position and graph start time, in milliseconds
+ * @param {Number} cursorOffset
+ * @return {Object} - Redux action
+ */
+const setIVGraphCursorOffset = function(cursorOffset) {
+    return {
+        type: 'SET_IV_GRAPH_CURSOR_OFFSET',
+        cursorOffset
+    };
+};
+
+/*
+ * Synchronous action sets the IV graph brush offset - in milliseconds
+ * @param {Number} startOffset - difference between brush start and the start time of the displayed time series
+ * @param {Number} endOffset - difference between the end of the displayed time series and the brush end
+ * @return {Object} - Redux action
+ */
+const setIVGraphBrushOffset = function(startOffset, endOffset) {
+    return {
+        type: 'SET_IV_GRAPH_BRUSH_OFFSET',
+        startOffset,
+        endOffset
+    };
+};
+
+/*
+ * Synchronous action to clear the IV graph brush offset
+ * return {Object} - Redux action
+ */
+const clearIVGraphBrushOffset = function() {
+    return {
+        type: 'CLEAR_IV_GRAPH_BRUSH_OFFSET'
+    };
+};
+
+/*
+ * Synchronous action add the tsRequestKeys to  iv time series loading
+ * @param {Array of String} tsRequestKeys
+ * @return {Object} - Redux actions
+ */
+const addIVTimeSeriesToLoadingKeys = function(tsRequestKeys) {
+    return {
+        type: 'ADD_IV_TIME_SERIES_TO_LOADING_KEYS',
+        tsRequestKeys
+    };
+};
+
+/*
+ * Synchronous action to remove the tsRequestKeys from iv time series loading
+ * @param {Array of String} tsRequestKeys
+ * @return {Object} - Redux action
+ */
+const removeIVTimeSeriesFromLoadingKeys = function(tsRequestKeys) {
+    return {
+        type: 'REMOVE_IV_TIME_SERIES_FROM_LOADING_KEYS',
+        tsRequestKeys
+    };
+};
+
+/*
+ * Synchronous action to set audible play ID
+ * @param {Number} audiblePlayId - id of the timer
+ * @return {Object} - Redux actions
+ */
+const ivTimeSeriesPlayOn = function(audiblePlayId) {
+    return {
+        type: 'IV_TIME_SERIES_PLAY_ON',
+        audiblePlayId
+    };
+};
+
+/*
+ * Synchronous action to remove the audible play ID
+ * @return {Object} - Redux actions
+ */
+const ivTimeSeriesPlayStop = function() {
+    return {
+        type: 'IV_TIME_SERIES_PLAY_STOP'
+    };
+};
+
+/*
+ * Asynchronous Redux action which starts audio play. If the play reaches the end
+ * of the time series it is set back to the beginning
+ * @param {Number} maxCursorOffset - epoch in milliseconds
+ * @return {Function}
+ */
+const startTimeSeriesPlay = function(maxCursorOffset) {
+    return function (dispatch, getState) {
+        let state = getState().ivTimeSeriesState;
+        if (state.ivGraphCursorOffset == null || state.ivGraphCursorOffset >= maxCursorOffset) {
+            dispatch(Actions.setIVGraphCursorOffset(0));
+        }
+        if (!state.audiblePlayId) {
+            let play = function () {
+                let newOffset = getState().ivTimeSeriesState.ivGraphCursorOffset + 15 * 60 * 1000;
```

Comment:
Just curious, what's the `15` for? Is that 15 minutes in milliseconds?

Reply:
Yes ... that's the length that the playbook moves every tick of the clock.

Reply:
Ah, that makes sense.